akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala

```diff
@@ -273,9 +306,10 @@ private[akka] trait HandleBackoff { this: Actor ⇒
       // use the BackoffSupervisor as sender
       context.parent ! msg
 
-    case msg ⇒ child match {
-      case Some(c) ⇒ c.forward(msg)
-      case None    ⇒ context.system.deadLetters.forward(msg)
+    case msg ⇒ (child, replyWhileStopped) match {
+      case (Some(c), _) ⇒ c.forward(msg)
+      case (_, Some(r)) ⇒ sender ! r
+      case _            ⇒ context.system.deadLetters.forward(msg)
     }
```

Comment:
I'd prefer this as two match to avoid creating tuples.
```
    case msg ⇒ child match {
       case Some(c) ⇒ c.forward(msg)
       case None => replyWhileStopped match {
         case Some(r) ⇒ sender ! r
         case None  ⇒ context.system.deadLetters.forward(msg)
       }
      }
```